src/sync/outgoingSyncUtils.js

```diff
@@ -51,16 +53,37 @@ export function generateSyncJson(database, settings, syncOutRecord) {
   } else {
     // Get the record the syncOutRecord refers to from the database
     const recordResults = database.objects(recordType).filtered('id == $0', recordId);
-    if (!recordResults || recordResults.length === 0) { // No such record
+    if (!recordResults || recordResults.length === 0) {
+      // No such record
       throw new Error(`${recordType} with id = ${recordId} missing`);
-    } else if (recordResults.length > 1) { // Duplicate records
+    } else if (recordResults.length > 1) {
+      // Duplicate records
       throw new Error(`Multiple ${recordType} records with id = ${recordId}`);
     }
     const record = recordResults[0];
 
     // Generate the appropriate data for the sync object to carry, representing the
     // record in its upstream form
-    syncData = generateSyncData(settings, recordType, record);
+    try {
+      syncData = generateSyncData(settings, recordType, record);
+    } catch (error) {
+      // There was an error with data, often a null object
+      const siteName = settings.get(SYNC_SITE_NAME);
+      const syncUrl = settings.get(SYNC_URL);
+      const originalMessage = error.message;
+      // Change error message to be helpful in bugsnag
+      error.message =
+        `SYNC OUT ERROR. siteName: ${siteName}, serverUrl: ${syncUrl}, ` +
+        `syncOutRecord.id: ${syncOutRecord.id}, storeId: ${storeId} changeType: ${changeType}, ` +
+        `recordType: ${recordType}, recordId: ${recordId}, message: ${originalMessage}`;
+      // Ping the error off to bugsnag
+      bugsnagClient.notify(error);
+
+      // Make a nicer message for users and throw it again.
+      error.message =
```

Comment:
From what I understand bugsnag 'syncs' errors, does .notify store it by reference or does a deep copy ? Since the message is changed, would bugsnag's error.message be modified ?

Reply:
That is definitely a possibility. I'll just make copies of the error object.